In [5]:
cd nlp_class/miseless_data

/Users/shinbeompyeon/nlp_class/miseless_data


In [14]:
pwd # 현재 경로 확인

'/Users/shinbeompyeon/nlp_class/miseless_data'

In [7]:
import pandas as pd

In [16]:
path = 'SNS데이터/SNS_8.xlsx'
df = pd.read_excel(path, encoding='utf-8',index_col=0, header=0)
df

SECTION            DATE  \
DOCID                               
204667176      카페  20190204000000   
204667205      카페  20190204000000   
204667233      카페  20190204000000   
204667268      카페  20190204000000   
204667314      카페  20190204000000   
204667342      카페  20190204000000   
204667366      카페  20190204000000   
204667381      카페  20190204000000   
204667403      카페  20190204000000   
204667423      카페  20190204000000   
204667440      카페  20190204000000   
204667461      카페  20190204000000   
204667512      카페  20190204000000   
204667545      카페  20190204000000   
204667570      카페  20190204000000   
204667754      카페  20190204000000   
204667772      카페  20190204000000   
204667794      카페  20190204000000   
204667817      카페  20190204000000   
204667836      카페  20190204000000   
204667943      카페  20190204000000   
204668182      카페  20190204000000   
204668199      카페  20190204000000   
204668224      카페  20190204000000   
204668246      카페  20190204000000   
204668270      카페  20190204000000   
204668296      카페  20190204000000   
204668495      카페  20190204000000   
204668518      카페  20190204000000   
204668581      카페  20190204000000   
...           ...             ...   
204606719      뉴스  20180526000000   
204606729      뉴스  20180526000000   
204606741      뉴스  20180526000000   
204606753      뉴스  20180526000000   
204606772      뉴스  20180526000000   
204606785      뉴스  20180526000000   
204606791      뉴스  20180526000000   
204606933      뉴스  20180526000000   
204606948      뉴스  20180526000000   
204606960      뉴스  20180526000000   
204606976      뉴스  20180526000000   
204606987      뉴스  20180526000000   
204606999      뉴스  20180526000000   
204607008      뉴스  20180526000000   
204607018      뉴스  20180526000000   
204607034      뉴스  20180526000000   
204607232      뉴스  20180526000000   
204607245      뉴스  20180526000000   
204607267      뉴스  20180526000000   
204607278      뉴스  20180526000000   
204607291      뉴스  20180526000000   
204607300      뉴스  20180526000000   
204607317      뉴스  20180526000000   
204607328      뉴스  20180526000000   
204607421      뉴스  20180526000000   
204607431      뉴스  20180526000000   
204607440      뉴스  20180526000000   
204607450      뉴스  20180526000000   
204607511      뉴스  20180526000000   
204607521      뉴스  20180526000000   

                                                       TITLE  \
DOCID                                                          
204667176                               구제역은 재난문자 보내면서 미세먼지는   
204667205                          미세먼지마스크 슈슈마스크 정말 빨아써도 되나요   
204667233  7차 진행 다이슨 무선청소기 V6 앱솔루트 헤파 정품 새상품 최저가 택포  추가3종...   
204667268          2019년02월04일월 반송자율방범대 기해년 설 명절 기간 자율방범순찰활동   
204667314                      만5살인 딸에게 고민을 털어놓습니다 어디서 살고 싶니   
204667342                                    Seal Strip ASSY   
204667366                                      아기 이름 너무 고민되요   
204667381                                   전력거래소 환경을 고려한 정책   
204667403       속초양양고속도로로 부상하는 신흥해양상권1층30평대무권리상가앞마당80평해변가대로변   
204667423        속초양양고속도로 부상하는 신흥해양상권1층30평대무권리상가앞마당80평해변가대로변   
204667440                                         2월 5일 날씨예보   
204667461                                       2월 5 6일 황사예보   
204667512                                     1월 5째주 연두반 이야기   
204667545                                    가을스튜디오 촬영날 에피소드   
204667570             푸고 신상품 출시더 크고 새로운 컬러의 빨대컵을 베페에서 먼저 만나요   
204667754               2월 5일화2월 9일토 간단 예보일본 기상청의 황사예보 포함입니다   
204667772                              미세먼지 심한 날 아들과 달고나 만들기   
204667794                                       2월 5일 타팍 갑시다   
204667817                             요즘 갑자기 편도염 고생하시는분 계십니까   
204667836          추운 겨울 미세먼지 심한날 아이들이랑 함께 놀기 좋은 부천 플레이아쿠아리움   
204667943                                       낮에 MTB라이딩 운동   
204668182                             설날 미세먼지 일시 나쁨아침 영하권 추위   
204668199                                               속초여행   
204668224                             유치원 생일답례품 미세먼지 마스크 어때요   
204668246     

In [19]:
path1 = 'SNS데이터/SNS_1.xlsx'
path2 = 'SNS데이터/SNS_2.xlsx'
path3 = 'SNS데이터/SNS_3.xlsx'
path4 = 'SNS데이터/SNS_4.xlsx'
path5 = 'SNS데이터/SNS_5.xlsx'
path6 = 'SNS데이터/SNS_6.xlsx'
path7 = 'SNS데이터/SNS_7.xlsx'
path8 = 'SNS데이터/SNS_8.xlsx'

sns1 = pd.read_excel(path1, encoding='utf-8',index_col=0, header=0)
sns2 = pd.read_excel(path2, encoding='utf-8',index_col=0, header=0)
sns3 = pd.read_excel(path3, encoding='utf-8',index_col=0, header=0)
sns4 = pd.read_excel(path4, encoding='utf-8',index_col=0, header=0)
sns5 = pd.read_excel(path5, encoding='utf-8',index_col=0, header=0)
sns6 = pd.read_excel(path6, encoding='utf-8',index_col=0, header=0)
sns7 = pd.read_excel(path7, encoding='utf-8',index_col=0, header=0)
sns8 = pd.read_excel(path8, encoding='utf-8',index_col=0, header=0)

sns8.head(3)

SECTION            DATE  \
DOCID                               
204667176      카페  20190204000000   
204667205      카페  20190204000000   
204667233      카페  20190204000000   

                                                       TITLE  \
DOCID                                                          
204667176                               구제역은 재난문자 보내면서 미세먼지는   
204667205                          미세먼지마스크 슈슈마스크 정말 빨아써도 되나요   
204667233  7차 진행 다이슨 무선청소기 V6 앱솔루트 헤파 정품 새상품 최저가 택포  추가3종...   

                                                     CONTENT  
DOCID                                                         
204667176  지금까지 미세먼지로 재난문자 받은적이 한번도 없는데 재난문자 항목중에 대기오염은 없...  
204667205  미세먼지마스크를 하나 사려고 하거든요 나름 괜찮은 마스크를 사려고 알아봤는데 티엘마...  
204667233  7차 진행 다이슨 무선청소기 V6 앱솔루트 헤파 정품 새상품 최저가 택포  추가3종...

# SNS 데이터 전처리 전략

- 일단 한번 살펴보기

In [20]:
sns1.info()
sns2.info()
sns3.info()
sns4.info()
sns5.info()
sns6.info()
sns7.info()
sns8.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 176873 entries, 204172453 to 204047053
Data columns (total 4 columns):
SECTION    176873 non-null object
DATE       176873 non-null int64
TITLE      176818 non-null object
CONTENT    176668 non-null object
dtypes: int64(1), object(3)
memory usage: 6.7+ MB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 156112 entries, 204277076 to 204172200
Data columns (total 4 columns):
SECTION    156112 non-null object
DATE       156112 non-null int64
TITLE      156100 non-null object
CONTENT    154467 non-null object
dtypes: int64(1), object(3)
memory usage: 6.0+ MB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 72036 entries, 204615611 to 204081711
Data columns (total 4 columns):
SECTION    72036 non-null object
DATE       72036 non-null int64
TITLE      72034 non-null object
CONTENT    71394 non-null object
dtypes: int64(1), object(3)
memory usage: 2.7+ MB
<class 'pandas.core.frame.DataFrame'>
Int64Index: 67609 entries, 204482854 to 204477485

- .info를 본 결과

> 모든 데이터의 형식이 같음을 알 수 있음
SECTION    86101 non-null object
DATE       86101 non-null int64
TITLE      86088 non-null object
CONTENT    85193 non-null object

- section, date에는 빈값이 없고, title, content에는 빈값이 조금있음

176873
156112
72036
67609
86101
134219
63193
21689

In [32]:
sns8.SECTION.count()

21689

In [33]:
sns1

SECTION            DATE  \
DOCID                               
204172453     블로그  20181102235900   
204172474     블로그  20181102235700   
204172506     블로그  20181102235400   
204172546     블로그  20181102235400   
204172587     블로그  20181102235200   
204172624     블로그  20181102235200   
204172654     블로그  20181102235000   
204172688     블로그  20181102234500   
204172725     블로그  20181102234400   
204172756     블로그  20181102234400   
204172797     블로그  20181102234400   
204172841     블로그  20181102234300   
204172891     블로그  20181102234200   
204172935     블로그  20181102234100   
204172978     블로그  20181102234100   
204173022     블로그  20181102234100   
204173061     블로그  20181102233900   
204173101     블로그  20181102233800   
204173143     블로그  20181102233400   
204173177     블로그  20181102233000   
204173220     블로그  20181102232900   
204173257     블로그  20181102232600   
204173286     블로그  20181102232600   
204173353     블로그  20181102232500   
204173390     블로그  20181102232400   
204173434     블로그  20181102232400   
204173461     블로그  20181102232200   
204173505     블로그  20181102232000   
204173539     블로그  20181102231800   
204173564     블로그  20181102231600   
...           ...             ...   
204045692     블로그  20180508163100   
204045738     블로그  20180508163100   
204045785     블로그  20180508163000   
204045826     블로그  20180508163000   
204045866     블로그  20180508163000   
204045905     블로그  20180508162900   
204045961     블로그  20180508162900   
204046004     블로그  20180508162800   
204046055     블로그  20180508162700   
204046112     블로그  20180508162700   
204046153     블로그  20180508162600   
204046194     블로그  20180508162600   
204046240     블로그  20180508162600   
204046284     블로그  20180508162600   
204046326     블로그  20180508162600   
204046357     블로그  20180508162500   
204046391     블로그  20180508162500   
204046433     블로그  20180508162400   
204046488     블로그  20180508162300   
204046538     블로그  20180508162300   
204046584     블로그  20180508162200   
204046674     블로그  20180508162100   
204046729     블로그  20180508162000   
204046776     블로그  20180508162000   
204046810     블로그  20180508162000   
204046864     블로그  20180508162000   
204046921     블로그  20180508161900   
204046976     블로그  20180508161900   
204047012     블로그  20180508161900   
204047053     블로그  20180508161800   

                                                       TITLE  \
DOCID                                                          
204172453                        신동탄 이안파밀리에 눈에 보이는 프리미엄 진짜네요   
204172474                                           느린  점심點心   
204172506                     한화 태양의 숲 조성으로 미세먼지도 막고 환경도 지켜요   
204172546  섬유유연제추천 미세먼지까지 잡아 주는 착한 샤프란 섬유유연제 quot; 꽃담초 quot;   
204172587                            핑거루트 허브 다이어트 남자 몸매관리 후기   
204172624                                 어메의 280일 기록  임신26주   
204172654                                     당진국화축제와 서산국화축제   
204172688              삼성전자 에어드레서 의류청정기는 미세먼지 심한 겨울철에 필수템이네요   
204172725                  코타키나발루 1월 날씨 여행옷차림 강수량우기 매월 기후 정리   
204172756                베이비템샤프란 향기를 말을 하는샤프란 꽃담초 수 섬유유연제 장미   
204172797                    유아마스크 별과모래 면마스크 매일 써서 재구입이 시급하다   
204172841                              방학동용달이사수유동미아동용달이사원룸이사   
204172891                            20181016_마린크루수원센터 왕선생님반   
204172935                    임신 33주 34주 35주 36주차 일상 대전미즈여성병원   
204172978                                 빨래 건조기 강추 이건 꼭 사야해   
204173022           아빠 쉬는날엔 프리미엄 절충형 유모차 미고미니플러스3를 타고 나들이 가요   
204173061                      렌지 후드 주방 환풍기 교체 작업은 이렇게 진행합니다   
204173101                                평택 맛집 추천 회식모임도 조선육회   
204173143                              LG퓨리케어 공기청정기 AS128VWA   
204173177                          유산소운동 실내싸이클 알고타자 ⓦ멜킨스핀바이크   
204173220                                만기환급형실비보험 효율적핵심구성안    
204173257       천안놀이학교 놀이가 배움이 되는 블루마린용곡스포츠센터 롯데스포츠센터를 소개합니다   
204173286                 포인트 딥 클린 클렌징 오일 피부 세정 가을철 피부관리해보세요   
204173353                   오늘밤 그대 모습에 반해버렸네 가뭄 끝 단비를 맞은 것처럼   
204173390     

In [34]:
sns2

SECTION            DATE                                  TITLE  \
DOCID                                                                      
204277076      카페  20180706000000                               다이어트       
204277119      카페  20180706000000                   미세먼지 수치가 좋을때는 티가 나네용   
204277165      카페  20180706000000                            네이버날씨뉴스에 댓글   
204277224      카페  20180706000000                          오늘 서울랜드 다녀왔어요   
204277290      카페  20180706000000            77 경희대학교 관광학과 OIP 3기 임아란 과제   
204277345      카페  20180706000000                                     까꿍   
204277388      카페  20180706000000                         밤 사이에 비 소식 있나요   
204277431      카페  20180706000000           77 경희대학교 국제학과 JSC 17기 백승우 과제   
204277474      카페  20180706000000                        미세먼지 있는날 vs 없는날   
204277522      카페  20180706000000                  나트랑 날씨 이정도 되면 감사 해야겠죠   
204277579      카페  20180706000000   ♥ 7월 2  7월 6일 사랑반 어린이들의 놀이 속 들여다보기 ♥   
204277631      카페  20180706000000                                세상에 공기가   
204277683      카페  20180706000000                          밤새 미세먼지 괜찮을까요   
204277727      카페  20180706000000                    2018년7월2주 주간계획 및 일정   
204277788      카페  20180706000000               요새 우리나라 하늘 구름 너무 멋지지 않아요   
204277843      카페  20180706000000                       간만에 라이딩  두근두근하네요   
204277897      카페  20180706000000                소사본동신축빌라 소사역 소새울역 매력있어♥   
204277948      카페  20180706000000                         오늘 오후 4시넘어서 하늘   
204277999      카페  20180706000000            77 경희대학교 경영학과 OIP 3기 강영웅 과제   
204278392      카페  20180706000000                   미세먼지 오늘도 창문열구 자두 될까요   
204278441      카페  20180706000000                               병풀효능 찿아요   
204278493      카페  20180706000000                                   쉽  깝   
204278546      카페  20180706000000                       계속 미세먼지 없없음 좋겠네요   
204278590      카페  20180706000000                               병풀효능 찿아요   
204278649      카페  20180706000000                고농도 미세먼지 담당자 알림 다운로드 공지   
204278701      카페  20180706000000                              병풀 효능 찿아요   
204278774      카페  20180706000000                        81감사일기 20180706   
204278812      카페  20180706000000                              병풀 효능 찿아요   
204278870      카페  20180706000000               교환받은 모젤 라돈 수치입니다 2018628   
204279173      카페  20180706000000                    미세먼지 없으니 세상 깨끗하네요ㅜㅜ   
...           ...             ...                                    ...   
204169467      카페  20180429000000                               책육아하는 요즘   
204169518      카페  20180429000000                        오늘 같은날 어디 가야할까요   
204169974      카페  20180429000000                          어제 후쿠오카 도착했어요   
204170019      카페  20180429000000                                 대전 파레트   
204170072      카페  20180429000000                         빨래 마당에 널어도 되겠죠   
204170120      카페  20180429000000                                씻고나가려고요   
204170164      카페  20180429000000             해운대 센텀 천일 스카이원 숲세권APT 일반분양   
204170220      카페  20180429000000                          오늘같은 날 아차산 등반   
204170283      카페  20180429000000                                   미세먼지   
204170338      카페  20180429000000              해운대 센텀 천일 스카이원숲세권APT 일반분양   
204170386      카페  20180429000000                         미세먼지 상당히 나쁨이네요   
204170459      카페  20180429000000                      미세먼지 오늘도 나쁨 ㅎ ㅡ ㅎ   
204170907      카페  20180429000000         센텀 천일 스카이원 숲세권APT 262830평 일반분양   
204170956      카페  20180429000000                           오늘 미세먼지 어떤가요   
204170995      카페  20180429000000                                 파주금촌45   
204171059      카페  20180429000000                    제주도 미세먼지 장난 아닙니다 ㅠㅠ   
204171114      카페  20180429000000                           올만에 가족여행 스타트   
204171175      카페  20180429000000                          실내 오디갈만한데 없나요   
204171215      카페  20180429000000                             오늘은 꼭 나가시길   
204171267      카페  20180429000000                         

In [35]:
sns3

SECTION            DATE                                     TITLE  \
DOCID                                                                         
204615611      뉴스  20180525000000          날씨 내일 한낮 초여름 더위강한 자외선·오존·미세먼지 주의   
204615630      뉴스  20180525000000                    주말 내내 황사 주의내일 서울 낮 29도   
204615644      뉴스  20180525000000  빅스 라비 여자플러스2 청일점 MC 합격점 재치입담패션리더 新 뷰티 기대   
204615674      뉴스  20180525000000                  날씨 주말 미세먼지 나쁨낮 30도 안팎 더위   
204615683      뉴스  20180525000000               박문옥 전남도의원 목포1선거구 후보 본격 표밭다져   
204615693      뉴스  20180525000000                 날씨 당분간 비 소식 없어요미세먼지 계속 나쁨   
204615704      뉴스  20180525000000                  오영식 코레일 사장 강원도 관광 활성화 챙겨   
204615716      뉴스  20180525000000              김종민 서울시장 왜 나왔냐고 한국당 박물관 보내려고   
204615727      뉴스  20180525000000                 날씨 내일 초여름 더위 전국 오존 농도 나쁨’   
204615734      뉴스  20180525000000           박남춘 인천시장 민주당 후보 초심 잃지 않고 끝까지 최선   
204615752      뉴스  20180525000000               날씨 주말 맑고 낮기온 30도 가까이미세먼지 나쁨   
204615772      뉴스  20180525000000             날씨 주말 더위에 미세먼지까지서울 29도 대구 31도   
204615788      뉴스  20180525000000             내일26일 본격적인 초여름 날씨 시작겨땀 분수 주의보   
204615816      뉴스  20180525000000         반복적으로 올라오는 여드름 고민 유발시키는 원인부터 파악해야   
204615827      뉴스  20180525000000                  미세먼지 유발 화력발전 신재생에너지로 잡는다   
204615840      뉴스  20180525000000           26일 오늘의 날씨대전·세종·충남 7월 한여름 더위 기승   
204615870      뉴스  20180525000000             대구경북TBC어린이집 경로당 미세먼지 사각 대책 시급   
204615889      뉴스  20180525000000                  날씨 주말에도 미세먼지 나쁨한여름 더위 기승   
204615908      뉴스  20180525000000                           포토 뿌연 서울하늘 답답해요   
204616035      뉴스  20180525000000                주말 낮기온 30도 안팎 초여름 더위미세먼지 ↑   
204616044      뉴스  20180525000000     내일날씨 한낮 더위 기승 서울 29도·대구31도일부 미세먼지 나쁨’   
204616063      뉴스  20180525000000                    날씨 주말 30도 안팎 더위미세먼지 나쁨   
204616076      뉴스  20180525000000            내일날씨전국 맑고 일교차 커 수도권 미세먼지 농도 나쁨   
204616088      뉴스  20180525000000                           허태정 후보 정책공약집 발간   
204616103      뉴스  20180525000000                       내일날씨 전국 맑지만 미세먼지 나쁨   
204616149      뉴스  20180525000000                    날씨주말 전국 한여름 더위미세먼지 보통’   
204616160      뉴스  20180525000000        내일날씨 서울 낮 29도 등 전국 무더위수도권 미세먼지 나쁨’   
204616183      뉴스  20180525000000             내일 날씨 전국 맑고 더워수도권 미세먼지 나쁨’26일   
204616204      뉴스  20180525000000                   주말 낮동안 초여름 더위곳곳 미세먼지 주의   
204616366      뉴스  20180525000000                      연합뉴스 이 시각 헤드라인  2000   
...           ...             ...                                       ...   
204079693      뉴스  20180418000000   오늘의 날씨 낮 기온 19도 미세먼지 충북·영남 나쁨 예보 아침 안개주   
204079723      뉴스  20180418000000              날씨 맑지만 곳곳 미세먼지 나쁨’일교차 15도 안팎   
204079758      뉴스  20180418000000    기상청 오늘의 날씨 몰려오는 포근한 봄햇살 벌써 반팔 차림 경기남부    
204080215      뉴스  20180418000000                     버스 배기장치 개조 만연미세먼지 펑펑’   
204080249      뉴스  20180418000000                        4월 18일 굿모닝MBN 주요뉴스   
204080276      뉴스  20180418000000     tbs 김어준의 뉴스공장 김은경 환경부 장관 출연재활용 대란 대책은   
204080327      뉴스  20180418000000                  날씨 전국 맑고 포근 15도 내외 일교차 커   
204080352      뉴스  20180418000000               오늘날씨 오전 미세먼지 주의한낮 21도 맑고 포근   
204080500      뉴스  20180418000000               미세먼지 주범 공사장 먼지4곳 중 1곳 규정 위반   
204080527      뉴스  20180418000000               충북 큰 일교차 낮 기온 19∼21도미세먼지 나쁨   
204080559      뉴스  20180418000000                 경남도 친환경 도시숲’ 확대 조성에 발 벗었다   
204080858      뉴스  20180418000000              아침신문보기 윤리를 버린 기업 시장의 버림 받다 外   
204080886      뉴스  20180418000000              오늘 날씨 수도권 미세먼지 한때 나쁨서울 낮 20도   
204080918      뉴스  20180418000000               오늘 날씨 낮 20도 넘는다 아침 쌀쌀 낮엔 덥다   
204080945      뉴스  20180418000000                         오늘의 날씨 충북·세종18일 수   
204080987      뉴스  20180418000000                  날씨 전국 대체로 맑음오늘도 미세먼지 농도↑   
204081048      뉴스  20180418000000   김어준의 뉴스공장 노회찬 4월 국회 집어삼킨 드루킹김기식 사태에 대한    
204081076      뉴스  20180418000000                           

In [36]:
sns4

SECTION            DATE  \
DOCID                               
204482854      뉴스  20180806000000   
204484589      뉴스  20180806000000   
204469815      뉴스  20180801000000   
204416843     블로그  20190318235900   
204416886     블로그  20190318235900   
204416938     블로그  20190318235900   
204416995     블로그  20190318235900   
204417029     블로그  20190318235900   
204417073     블로그  20190318235900   
204417119     블로그  20190318235900   
204417152     블로그  20190318235800   
204417197     블로그  20190318235800   
204417229     블로그  20190318235800   
204417261     블로그  20190318235800   
204417298     블로그  20190318235800   
204417382     블로그  20190318235800   
204417423     블로그  20190318235800   
204417456     블로그  20190318235800   
204417501     블로그  20190318235700   
204417546     블로그  20190318235700   
204417578     블로그  20190318235700   
204417620     블로그  20190318235700   
204417660     블로그  20190318235700   
204417689     블로그  20190318235700   
204417722     블로그  20190318235600   
204417755     블로그  20190318235600   
204417786     블로그  20190318235600   
204417828     블로그  20190318235600   
204417858     블로그  20190318235500   
204417888     블로그  20190318235500   
...           ...             ...   
204475502      카페  20180910000000   
204475579      카페  20180910000000   
204475734      카페  20180910000000   
204475755      카페  20180910000000   
204475774      카페  20180910000000   
204475790      카페  20180910000000   
204475807      카페  20180910000000   
204475830      카페  20180910000000   
204475850      카페  20180910000000   
204475872      카페  20180910000000   
204475914      카페  20180910000000   
204476227      카페  20180910000000   
204476260      카페  20180910000000   
204476477      카페  20180910000000   
204476506      카페  20180910000000   
204476561      카페  20180910000000   
204476615      카페  20180910000000   
204476671      카페  20180910000000   
204476857      카페  20180910000000   
204476888      카페  20180910000000   
204476906      카페  20180910000000   
204476933      카페  20180910000000   
204476983      카페  20180910000000   
204477018      카페  20180910000000   
204477205      카페  20180910000000   
204477227      카페  20180910000000   
204477258      카페  20180910000000   
204477280      카페  20180910000000   
204477303      카페  20180910000000   
204477485      카페  20180910000000   

                                                       TITLE  \
DOCID                                                          
204482854                                   8월 주요 대학병원 건강강좌1   
204484589                  강릉 폭우 기상청 지형적 특이성으로 인해 많은 비가 내린 것   
204469815                               홍천날씨 더워도 너무 덥다역대 최고치   
204416843                  기아 올뉴쏘렌토 2WD 20 노블레스  중고차시세지나치지마요   
204416886                                             201901   
204416938         청라 돼지국밥 lt;가야밀면gt; 청라 맛집 추천  추어국수얼큰돼지국밥 후기   
204416995                      영어잘하는법 쉬우면서도 까다로운 과거시제 쓰임과 형태   
204417029                                4계절이 뚜렷하고 3면이 바다로 된   
204417073     소니 대학생 리뷰어 알파 마스터즈 7주차  A7M3  SEL2470Z로 담은 봉은사   
204417119                            20대 남친 생일선물 마음표현하기엔 딱이죠   
204417152                   딥클렌징 해피바스 마이크로 25 폼클렌징으로 미세먼지 세안   
204417197           파주금촌 꿀벅지닭갈비  숯불닭갈비 닭반마리칼국수 닭반마리짬뽕 닭똥집 닭발   
204417229                             유리물때제거 유리막코팅까지 한번에 해결    
204417261                                               오한내할   
204417298  탁한 하늘에 지쳤던 어느날 푸른 바다를 만났다  Welcome to the East...   
204417382                  H2CO3 리얼 펩타이드 탄산 미스트팩 톡톡 터지는게 신기해   
204417423                        2019시즌 K리그 개막전 전주성 직관 다녀왔어요   
204417456                         나비엔 에어원 청정환기로 실내 공기질을 깨끗하게   
204417501  구례 산수유축제  광양 매화축제 방문기 반곡마을홍쌍리 청매실농원주차꿀팁걷는 인간청매...   
204417546                     JAJU K94 황사마스크 블랙 도트로 패션까지 잡았지   
204417578                             문제가 문제인가 문제라는 명제가 문제인가   
204417620          경기도 하남 요정출몰지역  강동송파 인근의 아이랑 동물체험할 수 있는 카페   
204417660               하이트진로 신제품 테라 맥주 terra 리얼 탄산  청정라거 리뷰   
204417689                               라돈 측정기 라돈아이로 라돈수치 체크   
204417722     

In [37]:
sns5

SECTION            DATE  \
DOCID                               
205138319      뉴스  20190313000000   
205138335      뉴스  20190313000000   
205138352      뉴스  20190313000000   
205138367      뉴스  20190313000000   
205138377      뉴스  20190313000000   
205138388      뉴스  20190313000000   
205138398      뉴스  20190313000000   
205138403      뉴스  20190313000000   
205138410      뉴스  20190313000000   
205138508      뉴스  20190313000000   
205138516      뉴스  20190313000000   
205138521      뉴스  20190313000000   
205138526      뉴스  20190313000000   
205138536      뉴스  20190313000000   
205138546      뉴스  20190313000000   
205138552      뉴스  20190313000000   
205138562      뉴스  20190313000000   
205138569      뉴스  20190313000000   
205138576      뉴스  20190313000000   
205138645      뉴스  20190313000000   
205138649      뉴스  20190313000000   
205138661      뉴스  20190313000000   
205138670      뉴스  20190313000000   
205138674      뉴스  20190313000000   
205138679      뉴스  20190313000000   
205138686      뉴스  20190313000000   
205138804      뉴스  20190313000000   
205138813      뉴스  20190313000000   
205138821      뉴스  20190313000000   
205138826      뉴스  20190313000000   
...           ...             ...   
204618726      카페  20190202000000   
204618744      카페  20190202000000   
204618761      카페  20190202000000   
204618784      카페  20190202000000   
204618818      카페  20190202000000   
204618841      카페  20190202000000   
204619021      카페  20190202000000   
204619054      카페  20190202000000   
204619076      카페  20190202000000   
204619095      카페  20190202000000   
204619119      카페  20190202000000   
204619150      카페  20190202000000   
204619173      카페  20190202000000   
204619195      카페  20190202000000   
204619306      카페  20190202000000   
204619326      카페  20190202000000   
204619346      카페  20190202000000   
204619371      카페  20190202000000   
204619398      카페  20190202000000   
204619503      카페  20190202000000   
204619639      카페  20190202000000   
204619653      카페  20190202000000   
204619674      카페  20190202000000   
204619694      카페  20190202000000   
204619717      카페  20190202000000   
204619745      카페  20190202000000   
204619769      카페  20190202000000   
204619791      카페  20190202000000   
204619825      카페  20190202000000   
204619850      카페  20190202000000   

                                                       TITLE  \
DOCID                                                          
205138319                                      뚜둥Tudung 뉴스링크   
205138335                            농산물 산지 폐기 그만수출 플랫폼 구축해야   
205138352            내가 제일 잘 나가 세븐일레븐·팔도·GS25 기업들의 핫한 제품은 무엇   
205138367                       경남도 환경정책위원회 개최 미세먼지·녹조 대책 자문   
205138377                         신재생에너지의 모든 것SWEET 2019’ 개막   
205138388                       초등 방과후학교 영어 수업 금지 1년 만에 부활한다   
205138398                                       포토 반갑다 파란 하늘   
205138403                        중부발전 협력사와 그린파트너십으로 미세먼지 잡는다   
205138410                          “밤마다 고깃집 냄새’ 코를 찔러 못살겠어요”   
205138508                     미세먼지 해결 LPG 차량 더 만들지 않으면 그림의 떡   
205138516              정병국 유치원 초중고 공기정화설비 의무화하는 학교보건법 개정안 국회   
205138521                         시민 안전만족도 조사’ 708 정책 긍정’ 평가   
205138526                                하이트진로 맥주 신제품 테라’ 출시   
205138536                                 장성군 고농도 미세먼지 대응 총력   
205138546                                 미세먼지도 줄이고 난방비도 아껴요   
205138552                문화향기따라 삶을 힐링하다 쓸데없이 꽃놀이를 꼭 가야 하는 이유   
205138562                        北매체 비핵화 입장 확고美당국자 과감한 결단 필요   
205138569                            백승민 5타점 삼성 kt에 시범경기 2연승   
205138576                             미세먼지 특보 시 건강취약계층 보호조치를   
205138645                    김관영 정치개혁 위해 靑 혁신 · 연동형 비례대표제 절실   
205138649                  김문수 “미세먼지 발생원 국내발생중국발북한발 미세먼지442”   
205138661                              한국석유관리원 석유담당공무원 교육 실시   
205138670              군포시 도시공원에 미세먼지 저감 도움되는 스트로브 잣나무 등 심는다   
205138674                 안산시 지역경제 활성화·일자리 창출 추경 2조3천835억 편성   
205138679     

In [38]:
sns6

SECTION            DATE  \
DOCID                               
205169613      뉴스  20181115000000   
205169653      뉴스  20181115000000   
205169662      뉴스  20181115000000   
205169666      뉴스  20181115000000   
205169671      뉴스  20181115000000   
205169677      뉴스  20181115000000   
205169689      뉴스  20181115000000   
205169697      뉴스  20181115000000   
205169704      뉴스  20181115000000   
205169715      뉴스  20181115000000   
205169722      뉴스  20181115000000   
205169729      뉴스  20181115000000   
205169732      뉴스  20181115000000   
205169739      뉴스  20181115000000   
205169751      뉴스  20181115000000   
205169772      뉴스  20181115000000   
205169777      뉴스  20181115000000   
205169780      뉴스  20181115000000   
205169823      뉴스  20181115000000   
205169831      뉴스  20181115000000   
205169837      뉴스  20181115000000   
205169843      뉴스  20181115000000   
205169847      뉴스  20181115000000   
205169850      뉴스  20181115000000   
205169853      뉴스  20181115000000   
205169858      뉴스  20181115000000   
205169865      뉴스  20181115000000   
205169910      뉴스  20181115000000   
205169920      뉴스  20181115000000   
205169959      뉴스  20181115000000   
...           ...             ...   
205089549      뉴스  20190215000000   
205089557      뉴스  20190215000000   
205089579      뉴스  20190215000000   
205089591      뉴스  20190215000000   
205089647      뉴스  20190215000000   
205089658      뉴스  20190215000000   
205089667      뉴스  20190215000000   
205089675      뉴스  20190215000000   
205089685      뉴스  20190215000000   
205089706      뉴스  20190215000000   
205089716      뉴스  20190215000000   
205089723      뉴스  20190215000000   
205089731      뉴스  20190215000000   
205089794      뉴스  20190215000000   
205089806      뉴스  20190215000000   
205089824      뉴스  20190215000000   
205089834      뉴스  20190215000000   
205089844      뉴스  20190215000000   
205089855      뉴스  20190215000000   
205089865      뉴스  20190215000000   
205089878      뉴스  20190215000000   
205089933      뉴스  20190215000000   
205089938      뉴스  20190215000000   
205089947      뉴스  20190215000000   
205089956      뉴스  20190215000000   
205089965      뉴스  20190215000000   
205090010      뉴스  20190215000000   
205090016      뉴스  20190215000000   
205090030      뉴스  20190215000000   
205090036      뉴스  20190215000000   

                                                 TITLE  \
DOCID                                                    
205169613                  수업 개선 복지 강화 한명한명 존중 제주교육 실현   
205169653                       내일 중국발 스모그 유입오후부터 쌀쌀해져   
205169662                   내일날씨 남부지방 약한 비 곳곳 미세먼지 나쁨’   
205169666                     영주국유림관리소 숲가꾸기 1일 체험행사 개최   
205169671                   발뮤다 공기청정기 실시간 검색어에 등장한 이유는   
205169677                       내일 날씨 미세먼지 나쁨남부지방 곳곳 비   
205169689  GS칼텍스 고유가 시대 주유비 인하와 발맞춰 리터당 100원 할인 받을 수 있   
205169697                    날씨 새벽부터 중국발 스모그 유입낮에는 찬바람   
205169704                      요즘 아파트 새 트렌드 미세먼지 저감 기술   
205169715      2018 경인 히트상품 기업체부문 대상생활가전300M 아가미 공기청정기   
205169722         오늘 미세먼지는 나쁨본격적인 겨울 추위 내일부터 시작 주말은 맑아   
205169729                여주 강천면 아강지모 SRF열병합발전소 결사반대 시위   
205169732      오늘 날씨 미세먼지는 나쁨 내일부터 본격적인 겨울 추위 시작주말 날씨는   
205169739          노후경유차 단속  미세먼지 절감 인천시 운행제한 단속시스템 설치   
205169751     공항철도 대한민국 기상산업대상 금상 이용객에 대기오염 등 정보 제공 성과   
205169772                      행감인물 기재위 민경선 도의원 민주당고양4   
205169777                             오산시 출산축하용품 추가 지원   
205169780     도내 시·군 중 빗물이용시설 적극 활용한 수원시 제23회 경기환경대상서    
205169823        수원시 장안구 라돈측정기 공유 서비스 등 실내공기질 관리 적극 추진   
205169831                        여야 경기·인천 의원 수험생 응원 열전   
205169837           인천시 2019년부터 친환경 전기버스 대중교통 노선버스에 투입   
205169843      수원시 스마트 레인시티 수원 제23회 경기환경대상 환경우수기관부문 경기   
205169847         날씨 내일 전국 대부분 초미세먼지 농도 나쁨’ 낮부터 기온 내려가   
205169850        연애의 참견 시즌2 주우재 떡볶이 코트 입고 동안외모 과시같이 봐요   
205169853      中 베이징 등 스모그 경보 최고조 국내 유입 시점에 관심↑미세먼지는 나   
205169858                      내일 날씨 전국 흐리고 곳곳 미세먼지 나쁨   
205169865               울산날씨중국 오염물질 습격울산 17일부터 미세먼지 나쁨

In [39]:
sns7

SECTION            DATE  \
DOCID                               
204468714      카페  20180921000000   
204468728      카페  20180921000000   
204468782      카페  20180921000000   
204468805      카페  20180921000000   
204468823      카페  20180921000000   
204468840      카페  20180921000000   
204468858      카페  20180921000000   
204468902      카페  20180921000000   
204468922      카페  20180921000000   
204468942      카페  20180921000000   
204468956      카페  20180921000000   
204468975      카페  20180921000000   
204468992      카페  20180921000000   
204469004      카페  20180921000000   
204469023      카페  20180921000000   
204469039      카페  20180921000000   
204469062      카페  20180921000000   
204469192      카페  20180921000000   
204469210      카페  20180921000000   
204469232      카페  20180921000000   
204469257      카페  20180921000000   
204469271      카페  20180921000000   
204469293      카페  20180921000000   
204469315      카페  20180921000000   
204469345      카페  20180921000000   
204469387      카페  20180921000000   
204469593      카페  20180921000000   
204469611      카페  20180921000000   
204469632      카페  20180921000000   
204469651      카페  20180921000000   
...           ...             ...   
204866544      카페  20180702000000   
204866567      카페  20180702000000   
204866731      카페  20180702000000   
204866757      카페  20180702000000   
204866792      카페  20180702000000   
204866815      카페  20180702000000   
204866852      카페  20180702000000   
204866877      카페  20180702000000   
204866913      카페  20180702000000   
204866944      카페  20180702000000   
204866967      카페  20180702000000   
204867001      카페  20180702000000   
204867140      카페  20180702000000   
204867160      카페  20180702000000   
204867190      카페  20180702000000   
204867227      카페  20180702000000   
204867251      카페  20180702000000   
204867274      카페  20180702000000   
204867301      카페  20180702000000   
204867320      카페  20180702000000   
204867341      카페  20180702000000   
204867366      카페  20180702000000   
204867516      카페  20180702000000   
204867541      카페  20180702000000   
204867575      카페  20180702000000   
204867594      카페  20180702000000   
204867613      카페  20180702000000   
204867642      카페  20180702000000   
204867689      카페  20180702000000   
204867712      카페  20180702000000   

                                                       TITLE  \
DOCID                                                          
204468714                               2차게시판 760번 글 이어서 보세요   
204468728                                        날이 금방 추워지네요   
204468782                                            내일 미세먼지   
204468805                                            안인화력발전소   
204468823                        20180921 미세먼지 센서로 미세먼지 측정하기   
204468840                        20180921 미세먼지 센서로 미세먼지 측정하기   
204468858                          천안시 광촉매’ 도로포장 기술로 미세먼지 저감   
204468902                              9루무 ♥ 9월21일_200원 행복실천   
204468922                                 9월 22일 토요일 미세먼지 예보   
204468942                                   20180920 대모산 산행기   
204468956                          LG 퓨리케어 공기청정기 28평 750000원   
204468975                            공식앱 핫신상 동물제습인형 굿아이템 입니다   
204468992                            동탄2 자전거길 오산리천 4교 연결 필요성   
204469004                                서평오늘도 대한민국은 이상기후입니다   
204469023  9월22일23일엘지vsKT 9월27일28일엘지vs기아1루 3루 프리미엄 테이블 블루...   
204469039  9월25일26일두산vs넥센9월29일30일두산vs엘지1루 3루 블루 레드 네이비 응원...   
204469062                   가족들 옹기종기·미세먼지는 가득 명절기간 눈병’ 조심하세요   
204469192  현대버스 일렉시티 유니버스 유니시티 슈퍼에어로시티 그린시티 에어로타운 카운티 마이티...   
204469210  중고퀄팀팀 가루흑흑 제발 지나가늨 그림쟁이님들 부디 저희팀 삽화에 드러와주세요ㅠㅠ ...   
204469232                    트맵 포기하고플 때 페널티 안 먹고 탈출하는 미세먼지 팁   
204469257  9월25일26일두산vs넥센9월29일30일두산vs엘지1루 3루 블루 레드 네이비 응원...   
204469271  9월22일23일엘지vsKT 9월27일28일엘지vs기아1루 3루 프리미엄 테이블 블루...   
204469293                                     아름다운 풍경이 예술 자수   
204469315                                    요즘 미세 먼지 좋음의 이유   
204469345     

In [40]:
sns8

SECTION            DATE  \
DOCID                               
204667176      카페  20190204000000   
204667205      카페  20190204000000   
204667233      카페  20190204000000   
204667268      카페  20190204000000   
204667314      카페  20190204000000   
204667342      카페  20190204000000   
204667366      카페  20190204000000   
204667381      카페  20190204000000   
204667403      카페  20190204000000   
204667423      카페  20190204000000   
204667440      카페  20190204000000   
204667461      카페  20190204000000   
204667512      카페  20190204000000   
204667545      카페  20190204000000   
204667570      카페  20190204000000   
204667754      카페  20190204000000   
204667772      카페  20190204000000   
204667794      카페  20190204000000   
204667817      카페  20190204000000   
204667836      카페  20190204000000   
204667943      카페  20190204000000   
204668182      카페  20190204000000   
204668199      카페  20190204000000   
204668224      카페  20190204000000   
204668246      카페  20190204000000   
204668270      카페  20190204000000   
204668296      카페  20190204000000   
204668495      카페  20190204000000   
204668518      카페  20190204000000   
204668581      카페  20190204000000   
...           ...             ...   
204606719      뉴스  20180526000000   
204606729      뉴스  20180526000000   
204606741      뉴스  20180526000000   
204606753      뉴스  20180526000000   
204606772      뉴스  20180526000000   
204606785      뉴스  20180526000000   
204606791      뉴스  20180526000000   
204606933      뉴스  20180526000000   
204606948      뉴스  20180526000000   
204606960      뉴스  20180526000000   
204606976      뉴스  20180526000000   
204606987      뉴스  20180526000000   
204606999      뉴스  20180526000000   
204607008      뉴스  20180526000000   
204607018      뉴스  20180526000000   
204607034      뉴스  20180526000000   
204607232      뉴스  20180526000000   
204607245      뉴스  20180526000000   
204607267      뉴스  20180526000000   
204607278      뉴스  20180526000000   
204607291      뉴스  20180526000000   
204607300      뉴스  20180526000000   
204607317      뉴스  20180526000000   
204607328      뉴스  20180526000000   
204607421      뉴스  20180526000000   
204607431      뉴스  20180526000000   
204607440      뉴스  20180526000000   
204607450      뉴스  20180526000000   
204607511      뉴스  20180526000000   
204607521      뉴스  20180526000000   

                                                       TITLE  \
DOCID                                                          
204667176                               구제역은 재난문자 보내면서 미세먼지는   
204667205                          미세먼지마스크 슈슈마스크 정말 빨아써도 되나요   
204667233  7차 진행 다이슨 무선청소기 V6 앱솔루트 헤파 정품 새상품 최저가 택포  추가3종...   
204667268          2019년02월04일월 반송자율방범대 기해년 설 명절 기간 자율방범순찰활동   
204667314                      만5살인 딸에게 고민을 털어놓습니다 어디서 살고 싶니   
204667342                                    Seal Strip ASSY   
204667366                                      아기 이름 너무 고민되요   
204667381                                   전력거래소 환경을 고려한 정책   
204667403       속초양양고속도로로 부상하는 신흥해양상권1층30평대무권리상가앞마당80평해변가대로변   
204667423        속초양양고속도로 부상하는 신흥해양상권1층30평대무권리상가앞마당80평해변가대로변   
204667440                                         2월 5일 날씨예보   
204667461                                       2월 5 6일 황사예보   
204667512                                     1월 5째주 연두반 이야기   
204667545                                    가을스튜디오 촬영날 에피소드   
204667570             푸고 신상품 출시더 크고 새로운 컬러의 빨대컵을 베페에서 먼저 만나요   
204667754               2월 5일화2월 9일토 간단 예보일본 기상청의 황사예보 포함입니다   
204667772                              미세먼지 심한 날 아들과 달고나 만들기   
204667794                                       2월 5일 타팍 갑시다   
204667817                             요즘 갑자기 편도염 고생하시는분 계십니까   
204667836          추운 겨울 미세먼지 심한날 아이들이랑 함께 놀기 좋은 부천 플레이아쿠아리움   
204667943                                       낮에 MTB라이딩 운동   
204668182                             설날 미세먼지 일시 나쁨아침 영하권 추위   
204668199                                               속초여행   
204668224                             유치원 생일답례품 미세먼지 마스크 어때요   
204668246     

- 전부 concat해서 붙여서 value_counts해봐야겠다

In [41]:
df_full = pd.concat([sns1,sns2,sns3,sns4,sns5,sns6,sns7,sns8],ignore_index=True)
# concat: 세로 합치기, merge: 가로 합치기
df = df_full.copy()

In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 777832 entries, 0 to 777831
Data columns (total 4 columns):
SECTION    777832 non-null object
DATE       777832 non-null int64
TITLE      777699 non-null object
CONTENT    772910 non-null object
dtypes: int64(1), object(3)
memory usage: 23.7+ MB


In [43]:
# # csv로파일 백업
# save_file = 'sns_full.csv'
# df.to_csv(save_file, index=False)

In [46]:
df['SECTION'].value_counts()

블로그    302903
카페     285658
뉴스     189271
Name: SECTION, dtype: int64

In [47]:
df['DATE'].value_counts()
# 그래프를 그려봐야겠다.

20190315000000    2603
20181203000000    2537
20181112000000    2496
20181221000000    2383
20190225000000    2378
20190111000000    2345
20190114000000    2315
20181109000000    2308
20181130000000    2274
20181128000000    2252
20190313000000    2218
20181022000000    2214
20190130000000    2214
20190302000000    2211
20181107000000    2183
20190220000000    2181
20190201000000    2145
20181219000000    2137
20190119000000    2136
20190316000000    2107
20190323000000    2097
20181224000000    2048
20181129000000    2021
20180618000000    1994
20180523000000    1957
20190121000000    1956
20190112000000    1911
20181117000000    1889
20190113000000    1878
20190319000000    1871
                  ... 
20180927142100       1
20180621171500       1
20180816231200       1
20180623235900       1
20180727061300       1
20190109145100       1
20180717140000       1
20180621162800       1
20180813101400       1
20180627231800       1
20190207195400       1
20181020224000       1
20180902115

In [48]:
df['TITLE'].value_counts()

# tokens, ngram 만들어봐야겠다. 일단 미세먼지 뒤에 'ㅠㅠ'가 많긴 하군.
# 미세먼지 키워드 엮어서 쓴 광고들도 최상위에 있음. -> 노이즈라고 볼수도, 마케팅에 이용할수도 있음
## 3위(563) 달 착륙은 왜 이렇게 많이 떴을까?, 미세먼지와 관련이 있나?
# 날씨나 출석, 안부를 물을 때 '미세먼지'라는 키워드가 엮이기도 한다. 미세먼지가 바꾼 안부법

미세먼지                                                       5188
달 뒷면에 착륙하려는 시도가                                             563
오늘의 날씨                                                      454
오늘 미세먼지                                                     332
미세먼지 좋음                                                     296
미세먼지ㅠㅠ                                                      284
미세먼지 ㅠㅠ                                                     283
오늘의날씨                                                       275
날씨                                                          272
미세먼지ㅜㅜ                                                      268
출석                                                          268
미세먼지 조심하세요                                                  259
미세먼지 나쁨                                                     251
특가 할인중 단기 대환영 수원 영통 영통역 5분 전국에서 가장 큰 방을 관리비 없이 33만원에        247
출석합니다                                                       239
미세먼지 수치                                 

In [49]:
df['CONTENT'].value_counts()
# tokens, ngram 만들어봐야겠다. 일단 미세먼지 뒤에 'ㅠㅠ'가 많긴 하군.

- R studio나 스파이더에 올려서 봐야하나? 직관적으로 데이터를 둘러보기에 불편한 느낌

In [50]:
# 각 칼럼별 null값 보기
df.isnull().sum()

SECTION       0
DATE          0
TITLE       133
CONTENT    4922
dtype: int64